doozerlib/cli/release_gen_payload.py

```diff
@@ -324,13 +319,17 @@ def _get_mirror_sources(self, latest_builds, mismatched_siblings):
                     if record.private:  # exclude embargoed images from the ocp[-arch] imagestreams
                         yellow_print(f"Omitting embargoed image {pullspecs[-1]}")
                     else:
-                        self.runtime.logger.info(f"Adding {arch} image {pullspecs[-1]} to the public mirroring list with imagestream tag {tag_name}...")
-                        mirroring.setdefault((arch, False), {})[tag_name] = mirroring_value
+                        mirroring_list = mirroring.setdefault((arch, False), {})
+                        if tag_name not in mirroring_list or payload_name:  # arch-specific image with explicit payload_name serves as an override (https://issues.redhat.com/browse/ART-2823)
```

Comment:
it took me a minute to puzzle out how this is an override. might this be clearer as:
```suggestion
                        if tag_name not in mirroring_list or payload_name:
                            # if multiple images in arch have the same tag, only explicit payload_name overwrites (https://issues.redhat.com/browse/ART-2823
```